### PR TITLE
stream cached restore response

### PR DESCRIPTION
**Pull request by snopoke**
@benrudolph 

@emord review buddy

### CODE REVIEW

**Comment by snopoke** [comment]
Should this go behind a flag to allow gradual rollout? Probably.

File: corehq/ex-submodules/casexml/apps/phone/restore.py [hunk 1]
```diff
@@ -510,7 +518,7 @@ def get_cached_payload(self):
             return
 
         if self.sync_log:
-            payload = self.sync_log.get_cached_payload(self.version)
+            payload = self.sync_log.get_cached_payload(self.version, stream=True)
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: This returns a File like object which streams the cached attachment contents from CouchDB.

Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/phone/restore.py]: sweet didn't realize couchdbkit had this

**Comment by benrudolph** [comment]
This makes sense to me. agree with the flag since i believe this will affect things that are not FileRestore (StringResponses that are getting data back from the cache). also might make sense to update the doc string since at the current state we can receive a stream, a string and file name

File: corehq/toggles.py [hunk 2]
```diff
@@ -375,3 +375,9 @@ def toggles_dict(username=None, domain=None):
      "because it has to be accompanied by a migration.)"),
     [NAMESPACE_DOMAIN],
 )
+
+STREAM_RESTORE_CACHE = StaticToggle(
+    'stream_cached_restore',
+    'Stream cached restore from couchdb',
+    [NAMESPACE_DOMAIN]
+)
```
Comment (dimagimon) [inline corehq/toggles.py]: W292 no newline at end of file

**Comment by benrudolph** [comment]
(test failure looks real)

**Comment by snopoke** [comment]
I was curious to see if this SyncLog caching ever got used so checked the apache logs and found that it does: https://docs.google.com/a/dimagi.com/spreadsheets/d/1V3RSfDAFYByT5ujPOJUq5XGjqYkQzac2KGVkM-I7ZN4/edit#gid=0

**Comment by snopoke** [comment]
@benrudolph :white_check_mark:

**Comment by benrudolph** [comment]
cool, +1 for the logs. good to merge. currently unable to though. also if i'm not online when you fix the merge, feel free to self merge to stop the cycle of merge conflicts

**Pull request merged**